activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -234,12 +233,14 @@ def perform_calculation(operation, column_name)
         distinct = distinct_value
 
         if operation == "count"
+          distinct = select_for_count if distinct_select?(select_for_count)
```

Comment:
Can you remove this line in favor of https://github.com/rails/rails/pull/35361#discussion_r259570692?

Reply:
Done!